app/src/app/present-round/present-round.component.ts

```diff
@@ -84,11 +84,14 @@ export class PresentRoundComponent implements OnInit {
         )
         return votingRightsPath
       }
-    )
-    this.controlledVotingRights.pipe(
-      tap(votingRightsPaths => {
-        this.hasVotingRight = votingRightsPaths.length && votingRightsPaths.length >= 0})
+    ).pipe(
+      take(1),
```

Comment:
Moc to nechápu, kde byl problém. Ale tohle take(1) udělá ten observe jenom jednorázově a když se pak změní to co to poslouchá, tak už se to neupdatne. To je chtěný?

Reply:
Tohle byl kód co dělal problém:
```
this.controlledVotingRights = combineLatest(
      this.db.list("landsraad/votingRights", ref => ref.orderByChild("controlledBy").equalTo(this.delegateId)).snapshotChanges(),
      this.db.list("landsraad/questions/", ref => ref.orderByChild("byDelegateId").equalTo(this.delegateId)).snapshotChanges(),
      (votingRightsSnapshots, questionsSnapshots) => {
        let questions: QuestionWithDbPath[] = questionsSnapshots.map(
          snapshot => {
          let question = snapshot.payload.val() as Question
          return {
            ...question,
            dbPath: `landsraad/questions/${snapshot.key}`
          }
        })
        let votingRightsPath: VotingRightWithQuestionPath[] = votingRightsSnapshots.map(
          (votingRightSnapshot) => {
            let votingRight = votingRightSnapshot.payload.val() as VotingRight
            let questionForVotingRight = questions.find(
              question => {
                return question["byVotingRightId"] === votingRightSnapshot.key && question["roundId"] === this.roundId
              })
            if (questionForVotingRight) {
              return {
                ...votingRight,
                dbPath: questionForVotingRight.dbPath
              }
            }
          }
        )
        return votingRightsPath
      }
    )
    this.controlledVotingRights.pipe(
      tap(votingRightsPaths => {
        this.hasVotingRight = votingRightsPaths.length && votingRightsPaths.length >= 0})
    ).subscribe()
    this.db.object(`rounds/${this.roundId}`).valueChanges().pipe(
      tap(
        (round: RoundInfoBasic) => {
          this.smallSize = round && round.size.toLowerCase() === "small"
        }
      )
    ).subscribe()
```

Moje teorie:
`this.controlledVotingRights` byly assigned pozdeji, po `this.hasVotingRight`, co znamenalo ze ten "dekretovy container", ktery ma tuhle podminku: `<ng-container *ngIf="smallSize && hasVotingRight">` byl rendered, ale bez toho aby existovali `this.controlledVotingRights`, coz znamena ze jednotlive "boxy" pro navrhy dekretu nebyly zobrazeny, protoze pro ne je "podminka": `*ngFor="let votingRight of controlledVotingRights | async"`.

Ted jsem to zmenila tak, aby `this.controlledVotingRights` a `this.hasVotingRight` byly assigned v jedne funkci, takze race condition by se nemela prihodit.

Jeste to vyzkousim kdyz vymazu to `take(1)`, mas pravdu ze tam to muzeme pipnout kolikrat chceme, kdyz se zmeni jedna z tech observable co sledujeme.